src/test/java/org/commonjava/util/partyline/AbstractJointedIOTest.java

```diff
@@ -251,4 +265,191 @@ public String call()
             return readingResult;
         }
     }
+
+    static final class OpenOutputStreamTask
+            extends IOTask
+            implements Callable<String>
+    {
+        private OutputStream stream;
+
+        protected OpenOutputStreamTask( JoinableFileManager fileManager, String content, File file )
+        {
+            super( fileManager, content, file, null, -1 );
+        }
+
+        protected OpenOutputStreamTask( JoinableFileManager fileManager, String content, File file, long waiting )
+        {
+            super( fileManager, content, file, null, waiting );
+        }
+
+        protected OpenOutputStreamTask( JoinableFileManager fileManager, String content, File file,
+                                        CountDownLatch controlLatch )
+        {
+            super( fileManager, content, file, controlLatch, -1 );
+        }
+
+        protected OpenOutputStreamTask( JoinableFileManager fileManager, String content, File file,
+                                        CountDownLatch controlLatch, long waiting )
+        {
+            super( fileManager, content, file, controlLatch, waiting );
+        }
+
+        @Override
+        public void run()
+        {
+            try
+            {
+                stream = fileManager.openOutputStream( file );
+            }
+            catch ( final Exception e )
+            {
+                e.printStackTrace();
+                fail( "Failed to open stream: " + e.getMessage() );
+            }
+
+            if ( stream != null && waiting > -1 )
```

Comment:
It seems like using the Thread.sleep() style of waiting is part of what makes these tests a bit fragile. I was hoping that by switching to Byteman, we could setup thread gates using things like rendezvous() and waitFor()/signalWake(), or even the join functions. This would allow us to avoid time-sensitive sleep periods that can be hard to predict in CI when there are a ton of builds running.

For reference: http://downloads.jboss.org/byteman/3.0.6/ProgrammersGuide.html#thread-coordination-operations

Converting from the old method of using timing delays to using byteman to orchestrate the timing means you'll have to really understand what order the different operations / activities are supposed to happen...for each test.